# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/Http2Session.h
# Commit: a92c423736e5
# Full Hash: a92c423736e5979a8cf039017fd707d41f665b9c
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2020-05-25 15:42:48
# Regressor Bug: 1546358
# File Overlap Count: 1
# Description:
#   Bug 1546358 - Forward cache entry check to parent process r=dragana
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D75941
# ==============================================================================

diff -r b21cc935a336 -r a92c423736e5 netwerk/protocol/http/Http2Session.h
--- a/netwerk/protocol/http/Http2Session.h	Fri May 22 13:37:54 2020 +0000
+++ b/netwerk/protocol/http/Http2Session.h	Mon May 25 11:07:07 2020 +0000
@@ -564,22 +564,6 @@
 
   uint32_t mCntActivated;
 
-  class CachePushCheckCallback final : public nsICacheEntryOpenCallback {
-   public:
-    CachePushCheckCallback(Http2Session* session, uint32_t promisedID,
-                           const nsACString& requestString);
-
-    NS_DECL_ISUPPORTS
-    NS_DECL_NSICACHEENTRYOPENCALLBACK
-
-   private:
-    ~CachePushCheckCallback() = default;
-
-    RefPtr<Http2Session> mSession;
-    uint32_t mPromisedID;
-    nsHttpRequestHead mRequestHead;
-  };
-
   // A h2 session will be created before all socket events are trigered,
   // e.g. NS_NET_STATUS_TLS_HANDSHAKE_ENDED and for TFO many others.
   // We should propagate this events to the first nsHttpTransaction.